phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -4957,12 +4964,22 @@ private void returnAllSequences(ConcurrentMap<SequenceKey,Sequence> sequenceMap)
     public void addConnection(PhoenixConnection connection) throws SQLException {
         if (returnSequenceValues || shouldThrottleNumConnections) {
             synchronized (connectionCountLock) {
-                if (shouldThrottleNumConnections && connectionCount + 1 > maxConnectionsAllowed){
-                    GLOBAL_PHOENIX_CONNECTIONS_THROTTLED_COUNTER.increment();
-                    throw new SQLExceptionInfo.Builder(SQLExceptionCode.NEW_CONNECTION_THROTTLED).
-                        build().buildException();
+
+                if(shouldThrottleNumConnections) {
```

Comment:
The value of `shouldThrottleNumConnections` should now be set based on this internal conn config and the max num conns from a client right?

Reply:
Good point, added.